app/graphql/types/query_type.rb

```diff
@@ -3,12 +3,23 @@ class Types::QueryType < Types::BaseObject
     description "MojoTech office locations"
   end
 
+  field :location, Types::LocationType do
+    description "Target location to receive updates"
+    argument :city_name, String, "City name of location", required: true
+  end
+
   def locations
     Rails.cache.fetch('locations-response', expires_in: 15.minutes) do
       Location.all
     end
   end
 
+  def location(city_name)
+    Rails.cache.fetch('location-response', expires_in: 15.minutes) do
+      Location.where(city_name: city_name[:city_name]).first
```

Comment:
Nit: `city_name` here is actually a hash argument. It would be idiomatic to use the ruby keyword arguments instead:
```
def locatoin(city_name: nil)
```
this is equivalent to:
```
def location(options = {})
  city_name = options[:city_name]
```